internal/controllers/machineset/machineset_controller.go

```diff
@@ -427,10 +427,12 @@ func (r *Reconciler) reconcileDelete(ctx context.Context, s *scope) (ctrl.Result
 	// else delete owned machines.
 	for _, machine := range machineList {
 		if machine.DeletionTimestamp.IsZero() {
-			log.Info("Deleting Machine", "Machine", klog.KObj(machine))
 			if err := r.Client.Delete(ctx, machine); err != nil && !apierrors.IsNotFound(err) {
 				return ctrl.Result{}, errors.Wrapf(err, "failed to delete Machine %s", klog.KObj(machine))
 			}
+			// Note: We intentionally log after Delete because we want this log line to show up only after DeletionTimestamp has been set.
+			// Also, setting DeletionTimestamp doesn't mean the Machine is actually deleted (deletion takes some time).
+			log.Info("Deleting Machine (MachineSet deleted)", "Machine", klog.KObj(machine))
```

Comment:
Let's make ~ the same change to the MachineDeployment controller. The reconcileDelete code is almost identical. But I think we should also take a look at MachineSet creation logs

Reply:
I will prefer to keep this PR scoped to machine create and delete.
We can send a follow up PR for create and delete MS

Reply:
Sure